diff --git a/src/AssetLibrary.php b/src/AssetLibrary.php
--- a/src/AssetLibrary.php
+++ b/src/AssetLibrary.php
@@ -5,6 +5,7 @@ declare(strict_types=1);
 namespace Drupal\vite;
 
 use Drupal\Component\Utility\UrlHelper;
+use Drupal\Core\Extension\Exception\UnknownExtensionException;
 use Drupal\Core\Extension\ModuleExtensionList;
 use Drupal\Core\Extension\ThemeExtensionList;
 use Drupal\Core\Messenger\MessengerInterface;
@@ -166,13 +167,18 @@ class AssetLibrary {
   private function getViteSettingFromExtensionDefinition(string $setting): mixed {
     // Get extension definition.
     $extensionDefinition = [];
-    if ($this->modules->exists($this->extension)) {
-      $extensionDefinition = $this->modules->getExtensionInfo($this->extension);
-    }
-    elseif ($this->themes->exists($this->extension)) {
-      $extensionDefinition = $this->themes->getExtensionInfo($this->extension);
+    try {
+      if ($this->modules->exists($this->extension)) {
+        $extensionDefinition = $this->modules->getExtensionInfo($this->extension);
+      }
+      elseif ($this->themes->exists($this->extension)) {
+        $extensionDefinition = $this->themes->getExtensionInfo($this->extension);
+      }
+    }
+    catch (UnknownExtensionException) {
+      return NULL;
     }
 
     // Check if extension definition has vite settings.
     if (!isset($extensionDefinition['vite'])) {
       return NULL;
